### ISSUE TITLE
Invalid configuration mapping when loading an SDK-style project

### ISSUE THREAD

**Issue by KirillOsenkov** [mono/monodevelop#6335]
VSMac 7.7.0.1624.

When loading an SDK-style project with SDK `Xamarin.Mac.Sdk/0.12.0` the project fails to fully load and displays the `Invalid configuration mapping` message.

https://github.com/mono/monodevelop/blob/617c62128a87dd128697031315d279b55d38aaf5/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/ProjectNodeBuilder.cs#L148

This is because CSharpProject.Configurations is empty (and it expects `Debug`).

> VS bug [#706555](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/706555)

**Comment by KirillOsenkov** [comment]
I've added the repro steps here: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/706555

**Comment by KirillOsenkov** [comment]
OK @sandyarmstrong and I have figured out what is going on. 

`FindPotentialSdkResolvers()` is missing the logic to look for manifest .xml files:
https://github.com/mono/monodevelop/blob/220cdf3e78d416ee451af52f2133901a1f39429c/main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild/SdkResolution.cs#L139

The layout of the folder is:
`bin/SdkResolvers/Microsoft.Build.NuGetSdkResolver/Microsoft.Build.NuGetSdkResolver.xml` but VSMac logic doesn't take that into account.

See related:
https://github.com/Microsoft/msbuild/pull/3239

**Comment by KirillOsenkov** [comment]
The logic needs to be synchronized with the logic here:
https://github.com/Microsoft/msbuild/blob/e70a3159d64f9ed6ec3b60253ef863fa883a99b1/src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs#L45-L51

**Comment by abock** [comment]
Probably related: #6342

**Comment by sandyarmstrong** [comment]
Due to https://github.com/mono/mono/issues/11202, you may need to wipe out `/Library/Frameworks/Mono.framework/Versions/5.16.0` before you can repro. We have been testing with Mono 5.16.0.173.

**Comment by KirillOsenkov** [comment]
And you can use any SDK-style project that is using any custom SDK, e.g.

```
<Project Sdk="Xamarin.Mac.Sdk/0.12.0">

  <PropertyGroup>
    <TargetFramework>net461</TargetFramework>
  </PropertyGroup>

</Project>
```

**Comment by sandyarmstrong** [comment]
Presumably the SDK has to come from NuGet.

I believe this is the PR from the msbuild side: https://github.com/Microsoft/msbuild/pull/3239

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-25] Comment by **xamarinc**



Fixed in version 7.8.0.969 (master)



Pull Request #6387 merged by: Matt Ward  
 Author: mono  
 Commit: c813a1a1ebccf1a47b497d728e81ec10d3a95967 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-26] Comment by **xamarinc**



Fixed in version 7.7.0.1720 (release-7.7)



Pull Request #6391 merged by: Matt Ward  
 Author: mono  
 Commit: a6773f31a7f57d4600889363ac3acffa6707ba72 (mono/monodevelop)

**Comment by sandyarmstrong** [comment]
Thanks! I tested a 7.7 build with this fix and it seemed to work great for me!

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-30] Comment by **xamarinc**



Fixed in version 8.0.0.1738 (release-8.0)



Author: Matt Ward  
 Commit: 2c88def712ea05c9823773061f958b4f11769483 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-34] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #6387 merged by: Matt Ward  
 Author: mono  
 Commit: c813a1a1ebccf1a47b497d728e81ec10d3a95967 (xamarin/monodevelop)
